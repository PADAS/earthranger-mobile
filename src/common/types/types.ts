import { EventStatus, PatrolResult } from '../enums/enums';

export type Position = [number, number];

export type RootStackParamList = {
  SplashScreen: undefined,
  LoginView: undefined,
  TrackLocationMapView: {
    flyTo?: Position;
  }
  MainTabBar: undefined,
  ReportsView: undefined,
  ReportIssueView: undefined,
  PinAuthenticationView: undefined,
  SyncLoaderView: undefined,
  ReportCategoriesView: {
    coordinates?: Position,
  },
  ReportTypesView: {
    title: string,
    categoryId: string,
    coordinates?: Position,
    isPatrolInfoEventType?: boolean,
  },
  ReportForm: {
    reportId?: number,
    title: string,
    typeId: string,
    coordinates?: Position
    schema: string,
    geometryType: string,
    isEditMode?: boolean,
    createdAt?: string,
    isDefaultPatrolInfoEnabled?: boolean,
    geoJson?: GeoJSON.FeatureCollection,
    areaMeters?: number,
    perimeterMeters?: number,
    polygonPoints?: string,
    viewBox?: string,
    isPatrolInfoEvent?: boolean,
  },
  ReportEditLocationView: { coordinates: Position },
  RecordReportAreaView: undefined,
  AboutView: undefined,
  ReportNoteView: { id: number, text: string, coordinates: Position, channelId: string },
  MenuSettingsView: undefined,
  SettingsView: undefined,
  StatusView: undefined,
  ReportRepeatableFieldListView: {
    schema: any,
    title: string,
    rendererId: string,
    data?: RepeatableFieldListData[],
  },
  RepeatableFormView: {
    schema: any,
    title: string,
    listId: string,
  },
  EventsListFilterView: undefined,
  ReportsPendingSync: undefined,
  TrackedBySubjectsView: {
    subjects?: any,
  },
  BasemapView: undefined,
  CoordinateUnitsView: undefined,
  PatrolsTypeView: {
    coordinates: Position,
  },
  StartPatrolView: {
    patrolTitle: string,
    patrolType: string,
    patrolTypeId: string,
    coordinates: Position,
  },
  PhotoQualityView: undefined,
  ResetDatabaseCacheView: undefined,
  SubjectsListView: {
    parentSubjectGroupName?: string,
    isParentView?: boolean,
    parentId?: string,
  }
};

export type BottomSheetParamList = {
  Basemap: undefined,
  PatrolDetailsView: { patrolId: number },
  PatrolTrack: undefined,
  TrackingStatusView: undefined,
  SubjectDetailsView: undefined,
};

export interface EdgeInsets {
  top: number;
  right: number;
  bottom: number;
  left: number;
}

export type RepeatableFieldListData = {
  id: number;
  title: string;
  data: any;
  hidden: boolean,
};

export type LastLocation = {
  timestamp: string;
  displayDate: string;
  latitude: number;
  longitude: number;
  accuracy: number;
};

export interface IconProps {
  width?: string;
  height?: string;
  color?: string;
  viewbox?: string;
  isDisabled?: boolean;
}

export type AttachmentPhotoType = 'photo';
export type AttachmentNoteType = 'note';
export type AttachmentUpdateNoteType = 'updateNote';
export type AttachmentAddNewNoteType = 'addNewNote';
export type AttachmentDeleteNoteType = 'deleteNote';

export type ReportFormEvent = {
  id: number;
  remoteId: string,
  accountId: string,
  eventTypeId: number;
  title: string;
  latitude: number;
  longitude: number;
  geometry: string;
  event_values: string;
  isDraft: number;
  createdAt: number,
  updatedAt: number,
  patrol_segment_id: number | undefined,
};

export type EventData = {
  eventTypeId: string;
  title: string;
  latitude: number;
  longitude: number;
  geometry: string;
  event_values: string;
  isDraft: number;
  createdAt: string,
  updatedAt: string,
  patrol_segment_id: number | undefined,
};

export type ReportFormFullEvent = {
  account_id: number;
  profile_id: number;
  event_type_id: string;
  title: string;
  latitude: number;
  longitude: number;
  geometry: string,
  event_values: string;
  is_draft: number;
  value: string;
  updated_at: string;
  created_at: string;
  id: string;
  geometry_type: string;
  segment_remote_id: string | undefined;
  patrol_segment_id: string | undefined;
};

export type PersistedEventType = {
  id?: number;
  remote_id: string;
  account_id: number;
  profile_id: string;
  value: string;
  display: string;
  schema: string;
  category_id: string;
  geometry_type: string;
  default_priority: string;
  icon: string;
  icon_svg: string;
};

export type PersistedPatrolType = {
  id: number;
  remote_id: string;
  account_id: number;
  value: string;
  display: string;
  icon: string;
  icon_svg: string;
  default_priority: number;
  is_active: number;
  is_selected: boolean;
};

export type PersistedPatrol = {
  id: number;
  remote_id: string;
  account_id: number;
  profile_id: number;
  title: string;
  priority: string;
  state: string;
  created_at: string;
  updated_at: string;
};

export type PersistedPatrolSegment = {
  id: number
  remote_id: string;
  patrol_id: number;
  patrol_type_id: number;
  start_latitude: string;
  start_longitude: string;
  stop_latitude: string;
  stop_longitude: string;
  start_time: string;
  end_time: string;
};

export type PersistedPatrolTypeSegment = {
  id: number;
  remote_id: string;
  patrol_id: number;
  patrol_type_id: number;
  start_latitude: string;
  start_longitude: string;
  stop_latitude: string;
  stop_longitude: string;
  start_time: string;
  end_time: string;
  patrol_type: string;
};

export type DraftReports = {
  id: number;
  title: string;
  type_id: number,
  updated_at: string;
  icon_svg: string;
  default_priority: number;
  schema: string,
  geometry_type: string,
  hidden?: boolean;
  is_draft?: boolean;
};

export type EventUploadDetails = {
  reportId: number;
  reportTitle: string;
  isReportUploaded: boolean;
  uploaded: {
    notes: number;
    images: number;
  };
  pending: {
    notes: number;
    images: number;
  };
  state: string;
};

export type ReportFormEventAttachments = {
  id: number,
  remote_id: string,
  profile_id: number,
  account_id: number,
  event_id: string;
  type: string;
  path: string;
  thumbnail_path: string,
  note_text: string;
  uploaded: number,
  attachment_id: string;
};

export type PersistedPatrolDetail = {
  remote_id: string,
  title: string,
  serial_number: string,
  start_latitude: number,
  start_longitude: number,
  start_time: number,
  icon_svg: number,
  created_at: number,
};

export type PatrolDetail = {
  remoteId: string,
  title: string,
  serialNumber: string,
  startLatitude: number,
  startLongitude: number,
  startTime: number,
  iconSVG: string
  startTimeFormatted: string,
  createdAt: number,
};

export type PersistedUserProfile = {
  id: number;
  remote_id: string;
  account_id: string;
  username: string;
  content_type: string;
  subject_id: string;
  pin: string;
  permissions: string,
};

export type UserInfo = {
  id: number,
  type: string,
  permissions: string,
};

export type PersistedUserSubject = {
  id: number;
  remote_id: string;
  account_id: string;
  name: string;
  content_type: string;
  isSelected: boolean;
};

export type Patrol = {
  id: number,
  typeId: string,
  type: string,
  display: string,
  coordinates: Position | null,
  status: PatrolResult,
  timestamp: string,
};

// Polygon Reducer

export type PolygonState = {
  past: Position[],
  current: Position[],
};

export type PolygonAction = {
  type: 'UNDO' | 'REDO' | 'ADD',
  value?: Position,
};

type TGeometry = 'polygon' | 'point';

export const Geometry: Record<TGeometry, string> = {
  polygon: 'polygon',
  point: 'point',
};

export type Size = {
  width: number,
  height: number,
};

export interface EventListItem {
  bgColor: string,
  defaultPriority: number,
  error: string,
  fgColor: string,
  hidden: boolean,
  icon: string,
  id: number,
  isDraft: boolean,
  labelText: string,
  remoteId: string,
  status: EventStatus,
  statusIcon: string,
  text: string,
  title: string,
  errorMsg?: string,
  hasErrorOnEvent: boolean,
}

export type SubjectGroupData = {
  count: number,
  isVisible: boolean,
  title: string,
  type: 'group',
  subjects: number[],
  id: string,
};

export type Subject = {
  id: number,
  name: string,
  type: 'subject',
  isVisible: boolean,
  icon: string,
  lastPosition: Position,
  lastPositionUpdate: string,
  isHidden: boolean,
};

export type SubjectGroupSyncItem = {
  name: string,
  remoteId: string,
  parentId?: string,
  parentLocalId?: string,
};

export type SubjectSyncItem = {
  iconSVG: string,
  isActive: string,
  lastPosition: string,
  lastPositionDate: string,
  name: string,
  remoteId: string,
  tracksAvailable: string,
  updatedAt: string,
};
