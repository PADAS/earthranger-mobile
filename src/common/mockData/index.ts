export const LOGIN_DATA = {
  siteName: '*',
  userName: '*',
  password: '*',
  rememberMe: true,
  loginEndPoint: 'https://develop.pamdas.org/oauth2/token/',
};

export const STORE_INITIAL_STATE = {
  session: {
    accessToken: '*',
    error: '',
  },
  reports: {
    reportDraftsCount: 0,
  },
};
