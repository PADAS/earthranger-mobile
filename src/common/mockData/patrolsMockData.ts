export const PATROL_TYPES_RESPONSE_MOCK_DATA = {
  data: [
    {
      default_priority: 200,
      display: 'Helicopter Patrol',
      icon_id: 'helicopter-patrol-icon',
      id: 'd0884b8c-4ecb-45da-841d-f2f8d6246abf',
      is_active: true,
      value: 'heli-p-new',
      ordernum: 1,
    },
  ],
  status: {
    code: 200,
    message: 'OK',
  },
};

export const PATROL_TRACKED_BY_RESPONSE_MOCK_DATA = {
  data: {
    $schema: 'http://json-schema.org/draft-04/schema#',
    dependencies: {},
    description: '',
    properties: {
      leader: {
        enum: [
          {
            additional: {
              country: '',
              region: '',
              rgb: '',
              sex: 'male',
              tm_animal_id: '',
            },
            common_name: null,
            content_type: 'observations.subject',
            country: '',
            created_at: '2022-08-12T10:49:04.519622-07:00',
            id: '46535104-c874-47b9-9ffa-1518c0e01e6e',
            image_url: '/static/car.svg',
            is_active: true,
            name: 'car',
            region: '',
            sex: 'male',
            subject_subtype: 'car',
            subject_type: 'vehicle',
            tracks_available: false,
            updated_at: '2022-08-12T10:49:04.519641-07:00',
            user: null,
          },
        ],
        enum_ext: [
          {
            title: 'car',
            value: {
              additional: {
                country: '',
                region: '',
                rgb: '',
                sex: 'male',
                tm_animal_id: '',
              },
              common_name: null,
              content_type: 'observations.subject',
              country: '',
              created_at: '2022-08-12T10:49:04.519622-07:00',
              id: '46535104-c874-47b9-9ffa-1518c0e01e6e',
              image_url: '/static/car.svg',
              is_active: true,
              name: 'car',
              region: '',
              sex: 'male',
              subject_subtype: 'car',
              subject_type: 'vehicle',
              tracks_available: false,
              updated_at: '2022-08-12T10:49:04.519641-07:00',
              user: null,
            },
          },
        ],
        required: false,
        title: 'Leader',
        type: 'object',
      },
    },
    required: [],
    type: 'object',
  },
  status: {
    code: 200,
    message: 'OK',
  },
};
