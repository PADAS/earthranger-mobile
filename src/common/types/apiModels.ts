export interface FileRequest {
  uri: string,
  type: string,
  name: string,
}

export enum ApiStatus {
  Unknown = 1,
  Succeeded = 200,
  BadRequest = 400,
  Unauthorized = 401,
  Forbidden = 403,
  NotFound = 404,
  ServerError = 500,
}
