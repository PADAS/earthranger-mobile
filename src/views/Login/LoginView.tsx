/* eslint-disable global-require */
// External Dependencies
import React, { useCallback, useEffect, useState } from 'react';
import {
  Image,
  Keyboard,
  KeyboardAvoidingView,
  TouchableWithoutFeedback,
  useWindowDimensions,
  View,
} from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { enablePromise } from 'react-native-sqlite-storage';
import { connect } from 'react-redux';
import { useTranslation } from 'react-i18next';
import NetInfo from '@react-native-community/netinfo';
import { isEmpty } from 'lodash-es';
import { CommonActions } from '@react-navigation/native';
import { ThunkDispatch } from 'redux-thunk/es/types';
import { Action } from 'redux';

// Internal Dependencies
import {
  ACTIVE_USER_NAME_KEY,
  ALERT_BUTTON_BACKGROUND_COLOR_BLUE,
  BASEMAP_KEY,
  EXPERIMENTAL_FEATURES_FLAG_KEY,
  IS_IOS,
  PARENT_USER_REMOTE_ID_KEY,
  SAVE_TO_CAMERA_ROLL,
  SCREEN_FIT_700_TO_600,
  SCREEN_FIT_LESS_THAN_600,
  TRACKED_BY_SUBJECT_STATUS_KEY,
  USER_NAME_KEY,
  USER_SUBJECT_NAME_KEY,
} from '../../common/constants/constants';
import { LOGIN_SCREEN } from '../../analytics/model/constantsScreens';
import { loginAction } from '../../common/state/actions/sessionActions';
import { setBoolForKey } from '../../common/data/storage/keyValue';
import { icons } from '../../ui/AssetsUtils';
import AnalyticsEvent, { analyticsEventToHashMap } from '../../analytics/model/analyticsEvent';
import { logEvent, logScreenView } from '../../analytics/wrapper/analyticsWrapper';
import { createLoginSuccessEvent } from '../../analytics/login/loginAnalytics';
import { screenViewEventToHashMap } from '../../analytics/model/analyticsScreenView';
import createScreenViewEvent from '../screenViewTracker/screenViewTracker';
import { LoginForm } from './components/LoginForm/LoginForm';
import { InitialState } from '../../common/types/redux';
import { getSession } from '../../common/data/storage/session';
import {
  getUserRemoteId, setUserId, useInitDatabase, useRemoveDB,
} from '../../common/data/PersistentStore';
import { getSecuredStringForKey, setSecuredStringForKey } from '../../common/data/storage/utils';
import { getRemoteUser } from '../../api/usersAPI';
import { CustomAlert } from '../../common/components/CustomAlert/CustomAlert';
import { LogOutAlertIcon } from '../../common/icons/LogOutAlertIcon';
import { usePopulateUsers } from '../../common/data/users/usePopulateUsers';
import { ApiStatus } from '../../common/types/apiModels';
import { logGeneral } from '../../common/utils/logUtils';
import { Loader } from '../../common/components/Loader/Loader';
import { getAuthState, setAuthState } from '../../common/utils/authUtils';
import { AuthState } from '../../common/enums/enums';
import { removeDBLocations } from '../../common/utils/locationUtils';
import { deleteSession } from '../../common/utils/deleteSession';

// Styles
import style from './LoginView.styles';

enablePromise(true);

// Constants
const MAX_VIEW_OFFSET = -70;
const MAIN_TAB_BAR = 'MainTabBar';
const SYNC_LOADER_VIEW = 'SyncLoaderView';
const PIN_AUTH_VIEW = 'PinAuthenticationView';

/**
 * Login View.
 *
 * @param Redux Props
 * @returns JSX
 */
const LoginView = ({
  navigation,
  route,
  accessToken,
  loginError,
  login,
}: any) => {
  // Hooks
  const { t } = useTranslation();
  const { height } = useWindowDimensions();
  const { removeDB } = useRemoveDB();
  const { initDatabase } = useInitDatabase();
  const { populateUsers, isPinWorkflowAvailable } = usePopulateUsers();

  // Component's State
  const { shouldShowTokenExpiredAlert } = route.params || { shouldShowTokenExpiredAlert: false };
  const [keyboardVerticalOffset, setKeyboardVerticalOffset] = useState(0);
  const [internetErrorMessage, setInternetErrorMessage] = useState('');
  const [fieldErrorOffset] = useState(0);
  const [connectionErrorOffset, setConnectionErrorOffset] = useState(0);
  const [tokenExpiredAlertVisible, setTokenExpiredAlertVisible] = useState(shouldShowTokenExpiredAlert || '');
  const [userAccountErrorMessage, setUserAccountErrorMessage] = useState('');
  const [isLoading, setIsLoading] = useState(false);

  // Utility Functions
  const trackAnalyticsEvent = useCallback(async (event: AnalyticsEvent) => {
    await logEvent(event.eventName, analyticsEventToHashMap(event));
  }, []);

  const trackScreenView = useCallback(async () => {
    await logScreenView(screenViewEventToHashMap(createScreenViewEvent(
      LOGIN_SCREEN,
      LOGIN_SCREEN,
    )));
  }, []);

  const updateKeyboardOffset = () => {
    let adjustment = 0;
    if (height <= 700 && height >= 600) adjustment = -height * (SCREEN_FIT_700_TO_600);
    else if (height < 600) adjustment = -height * (SCREEN_FIT_LESS_THAN_600);
    setKeyboardVerticalOffset(fieldErrorOffset + connectionErrorOffset + adjustment);
  };

  const navigateTo = (view: string) => {
    navigation.dispatch(CommonActions.reset({
      index: 0,
      routes: [{ name: view }],
    }));
  };

  // Component's Life-cycle
  useEffect(() => {
    trackScreenView();

    /* istanbul ignore next */
    NetInfo.addEventListener(async (state) => {
      if (state.isInternetReachable !== null) {
        setInternetErrorMessage(state.isInternetReachable ? '' : t('loginView.internetConnectionNeeded'));
      }
      if (isEmpty(internetErrorMessage)) {
        setConnectionErrorOffset(0);
      } else {
        setConnectionErrorOffset(MAX_VIEW_OFFSET);
      }

      updateKeyboardOffset();
    });
  }, []);

  useEffect(() => {
    const sessionDataFromSecureStorage = getSession();

    const setUserOrPopulateDB = async () => {
      // db up to date, set login user
      const userName = getSecuredStringForKey(USER_NAME_KEY);
      const user = await getRemoteUser(accessToken);

      if (userName) {
        setSecuredStringForKey(PARENT_USER_REMOTE_ID_KEY, user.data.id);
        const userRemoteId = await getUserRemoteId(userName);
        if (userRemoteId !== -1 && user && userRemoteId === user.data.id) {
          await setUserId(userName);
          const pinWorkflowAvailable = await isPinWorkflowAvailable();
          setAuthState(pinWorkflowAvailable ? AuthState.required : AuthState.notRequired);
          if (getAuthState() === AuthState.required) {
            setIsLoading(false);
            navigateTo(PIN_AUTH_VIEW);
            return;
          }
          setSecuredStringForKey(USER_NAME_KEY, userName);
          setSecuredStringForKey(ACTIVE_USER_NAME_KEY, userName);
          setSecuredStringForKey(USER_SUBJECT_NAME_KEY, userName);
          setIsLoading(false);
          navigateTo(MAIN_TAB_BAR);
        } else {
          setBoolForKey(EXPERIMENTAL_FEATURES_FLAG_KEY, false);
          setBoolForKey(TRACKED_BY_SUBJECT_STATUS_KEY, false);
          setBoolForKey(BASEMAP_KEY, false);
          setAuthState(AuthState.unknown);
          await removeDBLocations();
          await removeDB();
          await initDatabase();
          setSecuredStringForKey(USER_NAME_KEY, userName);
          setSecuredStringForKey(ACTIVE_USER_NAME_KEY, userName);
          setSecuredStringForKey(USER_SUBJECT_NAME_KEY, userName);
          setBoolForKey(SAVE_TO_CAMERA_ROLL, false);

          logGeneral.info(`Active User: ${userName}`);
          let apiStatus = ApiStatus.Unknown;
          try {
            apiStatus = await populateUsers(accessToken);
            if (apiStatus === ApiStatus.Succeeded) {
              setIsLoading(false);
              navigateTo(SYNC_LOADER_VIEW);
            } else {
              await deleteSession();
              setIsLoading(false);
              setUserAccountErrorMessage(t('loginView.userIssuesMessage'));
              return;
            }
          } catch (error) {
            logGeneral.error('[LoginView] Error while populating users', error);
            await deleteSession();
            setIsLoading(false);
            setUserAccountErrorMessage(t('loginView.userIssuesMessage'));
            return;
          }
        }
      }
      setIsLoading(false);
    };

    if (accessToken && sessionDataFromSecureStorage?.access_token) {
      trackAnalyticsEvent(createLoginSuccessEvent());
      setUserOrPopulateDB();
    }
  }, [accessToken]);

  useEffect(() => {
    if (loginError) {
      setIsLoading(false);
    }
  }, [loginError]);

  useEffect(() => {
    updateKeyboardOffset();
  }, [fieldErrorOffset, connectionErrorOffset]);

  return (
    <>
      <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
        <KeyboardAvoidingView
          behavior={IS_IOS ? 'padding' : 'height'}
          style={{ flex: 1 }}
          keyboardVerticalOffset={keyboardVerticalOffset}
        >
          <SafeAreaView style={style.containerSafeArea}>
            {/* Main Container */}
            <View style={style.container}>
              {/* Earth Ranger Logo */}
              <Image style={style.logo} source={icons.LoginView.loginLogo} />

              {/* Login Form */}
              <LoginForm
                loginError={loginError}
                internetErrorMessage={internetErrorMessage}
                trackAnalyticsEvent={trackAnalyticsEvent}
                login={login}
                userAccountErrorMessage={userAccountErrorMessage}
                onSubmit={() => setIsLoading(true)}
              />
            </View>
            {/* End Main Container */}
            <CustomAlert
              alertIcon={<LogOutAlertIcon />}
              displayAlert={tokenExpiredAlertVisible}
              alertTitleText={t('loginView.expiredTokenAlertDialog.title')}
              alertMessageText={t('loginView.expiredTokenAlertDialog.message')}
              positiveButtonText={t('loginView.expiredTokenAlertDialog.action')}
              onPositiveButtonPress={() => {
                setTokenExpiredAlertVisible(false);
              }}
              positiveButtonBackgroundColor={ALERT_BUTTON_BACKGROUND_COLOR_BLUE}
            />
          </SafeAreaView>
        </KeyboardAvoidingView>
      </TouchableWithoutFeedback>
      <Loader isVisible={isLoading} />
    </>
  );
};

const mapStateToProps = (state: InitialState) => ({
  accessToken: state.session.accessToken,
  loginError: state.session.error,
});

const mapDispatchToProps = (dispatch: ThunkDispatch<InitialState, void, Action>) => ({
  login: (
    usernameValue: string,
    passwordValue: string,
  ) => dispatch(loginAction(usernameValue, passwordValue)),
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginView);
