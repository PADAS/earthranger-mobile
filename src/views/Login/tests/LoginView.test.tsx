// External Dependencies
import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ReduxThunk from 'redux-thunk';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';

// Internal Dependencies
import LoginView from '../LoginView';
import { STORE_INITIAL_STATE } from '../../../common/mockData';
import { RootStackParamList } from '../../../common/types/types';

// Types + Interfaces
// eslint-disable-next-line max-len
type StackNavigationPropAlias = NativeStackNavigationProp<RootStackParamList, 'ReportTypesView'>;

describe('LoginView', () => {
  let store: any;
  let navigation: Pick<StackNavigationPropAlias, 'navigate', 'addListener'>;
  const route = {
    params: {
      shouldShowTokenExpiredAlert: false,
    },
  };

  beforeEach(() => {
    const middlewares = [ReduxThunk];
    const mockStore = configureMockStore(middlewares);
    store = mockStore(STORE_INITIAL_STATE);
    navigation = {
      navigate: jest.fn(),
      addListener: jest.fn(),
    };
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <Provider store={store}>
        <LoginView navigation={navigation} route={route} />
      </Provider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
