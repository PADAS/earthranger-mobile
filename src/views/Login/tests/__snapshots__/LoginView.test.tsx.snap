// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView matches snapshot 1`] = `
Array [
  <View
    accessible={false}
    focusable={true}
    onClick={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RNCSafeAreaView
      edges={
        Array [
          "bottom",
          "left",
          "right",
          "top",
        ]
      }
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "marginEnd": 18,
            "marginStart": 18,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/icons/ic-login-logo.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "height": 85,
              "marginBottom": 10,
              "marginTop": 21,
              "resizeMode": "contain",
              "width": 180,
            }
          }
        />
        <View>
          <Text
            style={
              Object {
                "color": "#888B8D",
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 22,
              }
            }
          >
            SITE NAME *
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              accessibilityLabel="Site Name"
              autoCapitalize="none"
              autoCorrect={false}
              blurOnSubmit={false}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              returnKeyType="next"
              style={
                Array [
                  Object {
                    "borderColor": "#63666A",
                    "borderRadius": 3,
                    "borderWidth": 0.5,
                    "height": 40,
                    "marginTop": 5,
                    "paddingStart": 8,
                    "width": "100%",
                  },
                  Object {
                    "flex": 1,
                    "marginEnd": 3,
                    "paddingStart": 8,
                  },
                  null,
                ]
              }
              testID="LoginView-SiteName"
              value=""
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#888B8D",
                    "fontSize": 12,
                    "fontWeight": "500",
                    "marginTop": 22,
                  },
                  Object {
                    "flex": 0,
                    "fontSize": 16,
                    "marginTop": 12,
                  },
                ]
              }
            >
              .pamdas.org
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#888B8D",
              "fontSize": 12,
              "fontWeight": "500",
              "marginTop": 22,
            }
          }
        >
          USER NAME *
        </Text>
        <TextInput
          accessibilityLabel="User Name"
          autoCapitalize="none"
          autoCorrect={false}
          blurOnSubmit={false}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          returnKeyType="next"
          style={
            Array [
              Object {
                "borderColor": "#63666A",
                "borderRadius": 3,
                "borderWidth": 0.5,
                "height": 40,
                "marginTop": 5,
                "paddingStart": 8,
                "width": "100%",
              },
              null,
            ]
          }
          testID="LoginView-Username"
          value=""
        />
        <View>
          <Text
            style={
              Object {
                "color": "#888B8D",
                "fontSize": 12,
                "fontWeight": "500",
                "marginTop": 22,
              }
            }
          >
            PASSWORD *
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              accessibilityLabel="Password"
              autoCapitalize="none"
              autoCorrect={false}
              blurOnSubmit={false}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              returnKeyType="next"
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "borderColor": "#63666A",
                    "borderRadius": 3,
                    "borderWidth": 0.5,
                    "height": 40,
                    "marginTop": 5,
                    "paddingStart": 8,
                    "width": "100%",
                  },
                  Object {
                    "paddingEnd": 55,
                    "paddingStart": 8,
                  },
                  null,
                ]
              }
              testID="LoginView-Password"
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              testID="LoginView-Password-Button"
            >
              <View
                style={
                  Array [
                    Object {
                      "position": "absolute",
                      "resizeMode": "contain",
                    },
                    Object {
                      "height": 15,
                      "right": 10,
                      "top": 17,
                      "width": 22,
                    },
                  ]
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="50"
                  bbWidth="50"
                  fill="#63666A"
                  focusable={false}
                  height="50"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 50,
                        "width": 50,
                      },
                    ]
                  }
                  vbHeight={48}
                  vbWidth={48}
                  width="50"
                >
                  <RNSVGGroup
                    fill={4284704362}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M11 4C13.76 4 16 6.24 16 9C16 9.65 15.87 10.26 15.64 10.83L18.56 13.75C20.07 12.49 21.26 10.86 21.99 9C20.26 4.61 15.99 1.5 10.99 1.5C9.59 1.5 8.25 1.75 7.01 2.2L9.17 4.36C9.74 4.13 10.35 4 11 4ZM1 1.27L3.28 3.55L3.74 4.01C2.08 5.3 0.78 7.02 0 9C1.73 13.39 6 16.5 11 16.5C12.55 16.5 14.03 16.2 15.38 15.66L15.8 16.08L18.73 19L20 17.73L2.27 0L1 1.27ZM6.53 6.8L8.08 8.35C8.03 8.56 8 8.78 8 9C8 10.66 9.34 12 11 12C11.22 12 11.44 11.97 11.65 11.92L13.2 13.47C12.53 13.8 11.79 14 11 14C8.24 14 6 11.76 6 9C6 8.21 6.2 7.47 6.53 6.8ZM10.84 6.02L13.99 9.17L14.01 9.01C14.01 7.35 12.67 6.01 11.01 6.01L10.84 6.02Z"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              Object {
                "marginTop": 23,
              },
            ]
          }
        >
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": false,
              }
            }
            accessible={true}
            pointerEvents="auto"
          >
            <RNCCheckbox
              accessibilityLabel="Remember Me"
              boxType="square"
              forwardedRef={null}
              onCheckColor="#FFFFFF"
              onFillColor="#0056C7"
              onValueChange={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "height": 32,
                    "width": 32,
                  },
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              testID="LoginView-RememberMe"
              tintColor="#888B8D"
              tintColors={
                Object {
                  "true": "#0056C7",
                }
              }
              value={false}
            />
          </View>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 16,
                "marginStart": 14,
              }
            }
          >
            Remember me
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
              "marginBottom": 34,
              "marginTop": "16%",
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#0056C7",
                  "borderRadius": 5,
                  "height": 48,
                  "justifyContent": "center",
                  "width": "100%",
                },
                Object {
                  "opacity": 0.5,
                },
              ]
            }
            testID="LoginView-LoginButton"
          >
            <Text
              accessibilityLabel="Log In"
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "letterSpacing": 0.25,
                  "lineHeight": 21,
                }
              }
            >
              Log In
            </Text>
          </View>
        </View>
      </View>
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        transparent={true}
        visible=""
      />
    </RNCSafeAreaView>
  </View>,
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />,
]
`;
